src/site/content/en/blog/webassembly-threads/index.md

```diff
@@ -0,0 +1,472 @@
+environments, where applications normally run I/O in synchronous, blocking manner. Check out the
+blog post about [Using asynchronous web APIs from WebAssembly](https://web.dev/asyncify/) if you'd
+like to learn more.
+
+Particularly in this case, the code synchronously invokes `pthread_create` to create a background
+thread, and follows up by another synchronous call to `pthread_join` that waits for the background
+thread to finish execution. However, Web Workers, that are used behind the scenes when this code is
+compiled with Emscripten, are asynchronous. So what happens is, `pthread_create` only _schedules_ a
+new Worker thread to be created on the next event loop run, but then `pthread_join` immediately
+blocks the event loop to wait for that Worker, and by doing so prevents it from ever being created.
+It's a classic example of a [deadlock](https://en.wikipedia.org/wiki/Deadlock).
+
+One way to solve this problem is to create a pool of Workers ahead of time, before the program has
+even started, so that by the time `pthread_create` is invoked, it only needs to take a ready-to-use
+Worker from the pool, run the provided callback on that background thread and return the Worker back
+to the pool. All of this can be done synchronously, so there won't be any deadlocks as long as the
+pool is sufficiently large.
+
+This is exactly what Emscripten allows with the [`-s
+PTHREAD_POOL_SIZE=...`](https://emsettings.surma.technology/#PTHREAD_POOL_SIZE) option. It allows to
+specify a number of threads - either a fixed number, or a JavaScript expression like
+[`navigator.hardwareConcurrency`](https://developer.mozilla.org/docs/Web/API/NavigatorConcurrentHardware/hardwareConcurrency)
+to create as many threads as there are cores on the CPU. The latter option is helpful when your code
+can scale to an arbitrary number of threads.
+
+In the example above, there is only one thread being created, so instead of reserving all cores it's
+sufficient to use `-s PTHREAD_POOL_SIZE=1`:
+
+```shell
+emcc -pthread -s PTHREAD_POOL_SIZE=1 example.c -o example.js
+```
+
+{% Aside 'success' %} This time, when you execute it, things work successfully: {% endAside %}
+
+
+```text
+Before the thread
+Inside the thread: 42
+After the thread
+Pthread 0x701510 exited.
+```
+
+{% Aside 'caution' %} There is another problem though: see that `sleep(1)` in the code example? It
+executes in the thread callback, meaning off the main thread, so it should be fine, right? Well, it
+isn't. {% endAside %}
+
+When `pthread_join` is called, it has to wait for the thread execution to finish, meaning that if
+the created thread is performing long-running tasks—in this case, sleeping 1 second—then the main
+thread will also have to block for the same amount of time till the results are back. When this JS
+is executed in the browser, it will block the UI thread for 1 second until the thread callback
```

Comment:
Should the "1 second" in the previous sentence be changed as well for symmetry?

Reply:
Actually, because this is a measurement it should be "1 second" per Google styleguide. @rachelandrew Here's the reference: https://developers.google.com/style/numbers?hl=en#numbers-as-numerals